File: server/Models/Recipe.cs
using System.ComponentModel.DataAnnotations;

namespace allspice.Models;

// NOTE CreatedFromRecipeId will be used to distinguish altered recipes from base recipes. If CreatedFromRecipeId is null it is a base recipe, if it is not null it is an altered recipe.
public class Recipe : RepoItem<int>
{
  [MaxLength(255)] public string Title { get; set; }
  [MaxLength(5000)] public string Instructions { get; set; }
  [Url, MaxLength(2000)] public string Img { get; set; }
  [MaxLength(255)] public string Category { get; set; }
  public int? CreatedFromRecipeId { get; set; }
  [MaxLength(255)] public string CreatorId { get; set; }
  public Profile Creator { get; set; }
}

File: client/src/models/Recipe.js
import { Profile } from "./Account.js";
import { DatabaseItem } from "./DatabaseItem.js";


export class Recipe extends DatabaseItem{
  /**
   * @typedef RecipeData
   * @property {number} id
   * @property {Date} createdAt
   * @property {Date} updatedAt
   * @property {string} title
   * @property {string} instructions
   * @property {string} img
   * @property {string} category
   * @property {number} createdFromRecipeId
   * @property {string} creatorId
   * @property {Profile} creator
   * @param {RecipeData} data 
   */
  constructor(data){
    super(data)
    this.title = data.title
    this.instructions = data.instructions
    this.img = data.img
    this.category = data.category
    this.createdFromRecipeId = data.createdFromRecipeId
    this.creatorId = data.creatorId
    this.creator = data.creator
  }
}